### DISCUSSION TITLE
Running Freeplane under Java 8 and Java 11 on Windows

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/freeplane#1647] [Announcements]
Preview 1.11.9_13 supports Java versions 8 and 11 to 21 on Windows. However, it comes at costs of extra launcher written in Java that fixes the command line parameters and occupies about 100 mb RAM extra.

I added the extra launcher layer because some user complained that Freeplane could not be executed under Java 11 any longer.

If you think it is not what we need I can drop support of Java 8 and Iava 11 on windows and remove the extra launcher.

Which way do you like more?

**Comment by i-plasm** [comment]
Hi Dimitry. Thank you for taking notice. Any of the alternatives would be fine in my opinion. I think the most important is that the user gets feedback in case the launch of the .exe fails, informing what went wrong and which Java versions are supported.

**Comment by dpolivaev** [comment]
Finally, I implemented my custom version of Launch4j so that no extra Java-based launcher is needed.

https://github.com/dpolivaev/launch4j
